types/mongodb/index.d.ts

```diff
@@ -625,13 +626,13 @@ export interface Collection<TSchema = Default> {
     /** @deprecated use updateOne, updateMany or bulkWrite */
     update(filter: Object, update: Object, options: ReplaceOneOptions & { multi?: boolean }, callback: MongoCallback<WriteOpResult>): void;
     //http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#updateMany
-    updateMany(filter: Object, update: Object, callback: MongoCallback<UpdateWriteOpResult>): void;
-    updateMany(filter: Object, update: Object, options?: { upsert?: boolean; w?: any; wtimeout?: number; j?: boolean; }): Promise<UpdateWriteOpResult>;
-    updateMany(filter: Object, update: Object, options: { upsert?: boolean; w?: any; wtimeout?: number; j?: boolean; }, callback: MongoCallback<UpdateWriteOpResult>): void;
+    updateMany<T = Object>(filter: Object, update: T, callback: MongoCallback<UpdateWriteOpResult>): void;
```

Comment:
Not sure I see the value of generics here. For some `T`, it takes a `T`. Unless `T` is used in more than one place, That's not any more better than just taking `Object`.

Reply:
@andy-ms I listed an example in the gist above. I short generics here will allow us to validate if we can cast DTO to model. Consider the following example:
```
function ctrl(dto: ITestDTO) {
  const test: ITestModel = dto // we can get rid of this line with generics for insert and update
   mongodb.collection('test').insertOne(dto) // here we want to make sure dto can be cast to ITestModel
}
```

Reply:
So the generics allow you to assert a type inside an expression... don't think that is worth changing the type definition. You could always use the identity function for a similar purpose: `function id<T>(value: T): T { return value; }`

Reply:
Ok. It's just not gonna be as beautiful. Thanks for your time